cryptoconditions/types/rsa.py

```diff
@@ -0,0 +1,240 @@
+    TYPE_ASN1_FULFILLMENT = 'rsaSha256Fulfillment'
+    TYPE_CATEGORY = 'simple'
+
+    COST_RIGHT_SHIFT = 6    # 2**6 = 64
+
+    def __init__(self, asn1=None):
+        super().__init__()
+        self.modulus = None
+        self.signature = None
+        self.asn1 = asn1
+
+    def parse_json(self, json):
+        self.modulus = urlsafe_b64decode(base64_add_padding(json['modulus']))
+        self.signature = urlsafe_b64decode(
+            base64_add_padding(json['signature']))
+
+    def parse_asn1_dict_payload(self, data):
+        self.modulus = data['modulus']
+        self.signature = data['signature']
+
+    @property
+    def fingerprint_contents(self):
+        """Produce the contents of the condition hash.
+
+        This function is called internally by the `getCondition` method.
+
+        Returns:
+            bytes: Encoded contents of fingerprint hash.
+
+        """
+        if self.modulus is None:
+            raise MissingDataError('Requires modulus')
+
+        asn1_obj = nat_decode({'modulus': self.modulus},
+                              asn1Spec=RsaFingerprintContents())
+        asn1_der = der_encode(asn1_obj)
+        return asn1_der
+
+    @property
+    def asn1_dict_payload(self):
+        return {'modulus': self.modulus, 'signature': self.signature}
+
+    def _set_public_modulus(self, modulus):
+        """Set the public modulus.
+
+        This is the modulus of the RSA public key. It has to be provided
+        as raw bytes with no leading zeros.
+
+        Args:
+            modulus (bytes): Public RSA modulus.
+
+        """
+        if not isinstance(modulus, bytes):
+            raise TypeError('Modulus must be bytes, was: {}'.format(modulus))
+
+        if modulus[0] == 0:
+            raise Exception('Modulus may not contain leading zeros')
+
+        if not 128 <= len(modulus) <= 512:
+            raise Exception(
+                'Modulus must be between 128 bytes (1017 bits) and ' +
+                '512 bytes (4096 bits), was: {} bytes'.format(len(modulus)))
+
+        self.modulus = modulus
+
+    def _set_signature(self, signature):
+        """Set the signature manually.
+
+        The signature must be a valid RSA-PSS siganture.
+
+        Args:
+            signature (bytes): RSA signature.
+
+        """
+        if not isinstance(signature, bytes):
+            raise TypeError('Signature must be bytes, was: ' + signature)
+
+        self.signature = signature
+
+    def sign(self, message, private_key):
+        """Sign the message.
+
+        This method will take the provided message and create a
+        signature using the provided RSA private key. The resulting
+        signature is stored in the fulfillment.
+
+        The key should be provided as a PEM encoded private key string.
+
+        The message is padded using RSA-PSS with SHA256.
+
+        Args:
+            message (bytes): Message to sign.
+            private_key (str):  RSA private key.
```

Comment:
I think it would be best to just expect the private key to be to be a byte string